rotkehlchen/chain/ethereum/manager.py

```diff
@@ -1198,3 +1210,250 @@ def get_basic_contract_info(self, address: ChecksumEthAddress) -> Dict[str, Any]
 
         self.contract_info_cache[address] = info
         return info
+
+    def update_curve_pools_cache(self) -> None:
+        # Delete current cache. Need to do this in case curve removes some pools
+        GlobalDBHandler().delete_requests_cache(cache_type=RequestsCacheType.CURVE_POOL_ADDR)
+        GlobalDBHandler().delete_requests_cache(cache_type=RequestsCacheType.CURVE_POOL_TOKENS)
+        # write new values to the cache
+        self.update_cure_official_pools_cache()
+        self.update_curve_metapools_cache()
+
+    def update_cure_official_pools_cache(self) -> None:
+        address_provider_contract = EthereumConstants.contract('CURVE_ADDRESS_PROVIDER')
+        registry_abi = EthereumConstants.abi('CURVE_REGISTRY')
+        get_registry_result = address_provider_contract.call(
+            ethereum=self,
+            method_name='get_registry',
+        )
+        registry_address = to_checksum_address(get_registry_result)
+        registry_contract = EthereumContract(
+            address=registry_address,
+            abi=registry_abi,
+            deployed_block=0,
+        )
+        registry_pool_count = registry_contract.call(
+            ethereum=self,
+            method_name='pool_count',
+        )
+        calls_to_registry_pools_list = []
+        for i in range(registry_pool_count):
+            calls_to_registry_pools_list.append(
+                (
+                    registry_contract.address,
+                    registry_contract.encode(
+                        method_name='pool_list',
+                        arguments=[i],
+                    ),
+                ),
+            )
+        registry_pools_result = multicall(
+            ethereum=self,
+            calls=calls_to_registry_pools_list,
+        )
+        pool_addresses = []
+        calls_to_registry_lp_tokens = []
+        calls_to_registry_coins = []
+        calls_to_registry_underlying_coins = []
+        for pool_addr_encoded in registry_pools_result:
+            decoded_pool_addr = hex_or_bytes_to_address(pool_addr_encoded)  # already checksumed
+            pool_addresses.append(decoded_pool_addr)
+            calls_to_registry_lp_tokens.append(
```

Comment:
I guess some parts of these pools updating functions could be abstracted but I decided to keep them as-is for better readability